recipes-graphics/drm/libdrm_2.4.102.imx.bb

```diff
@@ -69,5 +68,3 @@ RRECOMMENDS_${PN}-drivers_append_imxgpu = " ${PN}-vivante"
 FILES_${PN}-vivante = "${libdir}/libdrm_vivante.so.*"
 PACKAGECONFIG_append_imxgpu = " vivante"
 PACKAGECONFIG[vivante] = "-Dvivante=true,-Dvivante=false"
-
-PACKAGE_ARCH = "${MACHINE_SOCARCH}"
```

Comment:
Why you are dropping this? This can cause task checksum changes if we use non-i.MX machines or use mainline specific machines.

Reply:
I removed it because:
If I put `PREFERRED_VERSION_libdrm = "2.4.102.imx"` in local.conf and `MACHINE=qemuarm bitbake libdrm` the build fails because MACHINE_SOCARCH is not defined.

```
ERROR: libdrm-2.4.102.imx-r0 do_write_config: Execution of '/build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725' failed with exit
 code 1:
/build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725: line 150: /build/krm/fsl-community-bsp/build/tmp/work/-fslc-linux-gnueabi/li
bdrm/2.4.102.imx-r0/meson.cross: No such file or directory
WARNING: /build/krm/fsl-community-bsp/build/tmp/work/${MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/temp/run.do_write_config.58725:150 exit 1 from 'cat > /build/krm/fsl-community-bsp/build/tmp/work/$
{MACHINE_SOCARCH}-fslc-linux-gnueabi/libdrm/2.4.102.imx-r0/meson.cross  <<EOF
...
```
And, since libdrm is in the conf/machine/include/imx-base.inc I thought for the machines including that incfile PACKAGE_ARCH is set magically in the classes/fsl-dynamic-packagearch.bbclass.

But looking at it more closely there is an issue. PREFERRED_VERSION_mxX and the MACHINE_SOCARCH_FILTER_append_imxgpu use different overrides. So e.g. i.MX 6ULL will behave badly.
I will change the PREFERRED_VERSION to read PREFERRED_VERSION_libdrm_imxgpu ?= "2.4.102.imx".

Should I instead put a `COMPATIBLE_MACHINE = "(imxgpu)"`?

Reply:
Probably better to keep `PACKAGE_ARCH = "${MACHINE_SOCARCH}"` and add `COMPATIBLE_MACHINE = "(imxgpu)"`
Then we don't change the libdrm version used for i.MX machines not falling under `_imxgpu`.

Reply:
Perfect.